### DISCUSSION TITLE
What is the right way to load data to bottom sheet?

### DISCUSSION THREAD

**Discussion by herczegzoltan** [lucacivale/Maui.BottomSheet#8] [Q&A] [answered]
Originally I had content page and viewmodel. When the page triggered OnAppearing I called an async init task in the viewmodel to load the data. With bottom sheet there is no OnAppearing. What would be the right way to load async data?

**Comment by lucacivale** [comment, marked as answer]
You can either work with the Opening/Opened commands/events or load the data before you set IsOpen to true. This really depends on your UI and where you want to show a loading indicator.

**Comment by herczegzoltan** [comment]
Thank you, the OpenedCommand works as expected. I have another question: I would like to change the background color of the entire page. When I tried setting the background color of the bottom sheet, nothing changed.

To work around this, I added a Grid and changed its background color like this:

```
<bottomsheet:BottomSheet.ContentTemplate>
    <DataTemplate>
        <Grid BackgroundColor="Yellow" RowDefinitions="auto, auto, *"/>
    </DataTemplate>
</bottomsheet:BottomSheet.ContentTemplate>
```
This successfully changed the background color, but the area where the handle is located remains unchanged.

I then decided to hide the header and handle altogether and add a new row with an image to replicate the handle, allowing me to control the color.

I updated the bottom sheet configuration to remove the header and handle:

```
<bottomsheet:BottomSheet
    x:Class="MauiApp.BottomSheetVM"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:bottomsheet="clr-namespace:Maui.BottomSheet;assembly=Maui.BottomSheet"
    xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
    x:Name="this"
    HasHandle="False"
    HeaderAppearance="None"
    IsCancelable="True"
    IsDraggable="True"
    OpenedCommand="{Binding LoadCommand}"
    SelectedSheetState="Large"
    SheetStates="Large"
    ShowHeader="False">
```

However, when the bottom sheet loads with this setup, the handle is still visible. If I add a switch bound to a boolean value and toggle it on and off after the bottom sheet opens, the handle disappears, but I don't want it to be there by default. 

Could you suggest me how i could change the color of the background of the handle and the handle color as well or how to turn if off successfully by default?

The workaround I found when OpenCommand is called in the method I just set true and false
```
[RelayCommand]
private async Task LoadCommand() 
{
	HasHandle = true;
	HasHandle = false;
        // async data laoding
}
```

**Comment by lucacivale** [comment]
There is already an open issue regarding the BackgroundColor. I'll take care of it this weekend.

That the Handle is shown initially even if it's disabled is a new issue and I'll take care of it aswell.

Changing the color of the Handle itself isn't supported because this is only possible on Android. iOS doesn't expose an API for that.

**Comment by lucacivale** [comment]
@herczegzoltan Can you give me a full repro of your issue with the handle? In my tests the handle is not shown if `HasHandle=false`.
I fixed the issue with the BackgroundColor. Now you can set it on the BottomSheet as on every other View and it will affect the whole BottomSheet. I'll release a new version of the Plugin if I'm sure your handle issue is fixed aswell.

**Reply by herczegzoltan** [reply]
@lucacivale 
Steps to reproduce the handle issue.
1. Open BottomSheetVM.xaml and set HasHandle = false;
2. Start the app and click on "Open Bottom sheet" button.
3. **issue**: Handle shall not be there but it is there

Steps for workaround 
1. Open BottomSheetVM.xaml and set
`HasHandle="{Binding HasHandle}"`
2. Open BottomSheetVMViewModel.cs
uncomment the 	
```
//HasHandle = true;
//HasHandle = false;
```
inside `Opened` command to set true and false

3. run the application, the handle is set true and false when opening the bottomsheet -> no visible handle shown

https://github.com/herczegzoltan/TestForBottomSheet

Could you reproduce?

**Reply by herczegzoltan** [reply]
I have two more questions if you dont mind, I just did not found it. 
1. Is it possible to have custom header instead the right,left button and the title?
(obviously i could solve with Grid and create a header)

2.  Is it possible that the bottomsheet is only dragable using only the header?

**Reply by lucacivale** [reply]
I don't have access to your repository. Is it private?

1. No, currently there is not property for a custom header. Currently I'm working on a version for Maui v9. I'm updating the API a little bit to make it more convenient. Including more options to configure the header.

2. No, there is no native API to enable such behavior. Since this library relies on native implementations this is not possible.

**Reply by herczegzoltan** [reply]
Ohh sorry, just forgot to se to public now it shall work.

**Reply by lucacivale** [reply]
@herczegzoltan sorry for the late reply. I've checked your repro and I'm a bit confused. The default value of `BottomSheetVMViewModel.HasHandle` is true and therefore the handle is shown initially. This is correct behavior. If you want to hide the Handle you have to set `HasHandle = false`. If I do that in your report it works as expected.
